stm/stm.ml

```diff
@@ -2329,9 +2347,24 @@ end (* }}} *)
 (********************************* STM API ************************************)
 (******************************************************************************)
 
-let init () =
-  VCS.init Stateid.initial;
+type stm_init_options = {
+  doc_type     : stm_doc_type;
+(*
+  fb_handler   : Feedback.feedback -> unit;
+  iload_path   : (string list * string * bool) list;
+  require_libs : (Names.DirPath.t * string * bool option) list;
+  implicit_std : bool;
+*)
+}
+
+let doc_type_module_name (std : stm_doc_type) = match std with
+  | VoDoc  mn | VioDoc mn | Interactive mn -> mn
+
+let init { doc_type } =
+  VCS.init doc_type Stateid.initial;
   set_undo_classifier Backtrack.undo_vernac_classifier;
+  (* we declare the library here XXX: *)
+  let _mn = doc_type_module_name doc_type in
```

Comment:
Hum, I see this may help in a future PR, and cleaning things up is boring.  But why don't just comment this out?

Reply:
This was just to avoid the "unused warning".